[PM-9638] Browser V2 Item Details Defects

Jingo88 opened the pull request:
## 🎟️ Tracking

[PM-9638 Remove Owner](https://bitwarden.atlassian.net/browse/PM-9638)
[PM-9799 Update Organization Icons](https://bitwarden.atlassian.net/browse/PM-9799)
[PM-9808 Collection Word Wrap](https://bitwarden.atlassian.net/browse/PM-9808)

## 📔 Objective

- Removed excess Owner Element
- Updated Organization Icons to show `bwi-family` or `bwi-business` depending on org type
- Update styling for collections list
- Removed styling on item name - this should be added when we use the new display-field

## 📸 Screenshots

![Screenshot 2024-07-15 at 2 33 38 PM](https://github.com/user-attachments/assets/ba64d5f4-058b-41a2-aea3-4da07db88a1f)

![Screenshot 2024-07-15 at 2 33 22 PM](https://github.com/user-attachments/assets/c065ae87-456b-40a2-9a19-a1c991f0adbc)

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

<!-- Suggested interactions but feel free to use (or not) as you desire! -->

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

shane-melton reviewed (commented):
One possible suggestion if you want to take it up since this has come up a few times. Up to you though!

shane-melton commented on libs/angular/src/pipes/org-icon.pipe.ts:
```diff
@@ -0,0 +1,20 @@
+
+import { ProductTierType } from "@bitwarden/common/billing/enums";
+
+@Pipe({ name: "orgIcon" })
```
⛏️ We can make this a `standalone` pipe and then we can simply import it in the `item-details-v2.component` instead of `jslib`

shane-melton commented on libs/angular/src/pipes/org-icon.pipe.ts:
```diff
@@ -0,0 +1,20 @@
+
+@Pipe({ name: "orgIcon" })
+export class OrgIconPipe implements PipeTransform {
+  transform(productTierType: number): string {
```
❓ Can we use the `ProductTierType` here instead of `number` or was that giving you issues in the template?

shane-melton commented on libs/vault/src/cipher-view/item-details/item-details-v2.component.html:
```diff
@@ -16,18 +16,7 @@ <h2 bitTypography="h6">{{ "itemDetails" | i18n }}</h2>
-          }"
-        >
-        </i>
+        <i class="bwi bwi-lg bwi-fw tw-pt-1 {{ organization.productTierType | orgIcon }}"> </i>
```
💭 I know I said `pipe` in my original comment, but looking at this usage would it make more sense to have our own directive/component instead that takes care of the `bwi ...` classes.  They're almost always going to be used together. What do you think?

⛏️ We'll want to add an `aria-hidden` on this as well.

shane-melton commented on libs/vault/src/cipher-view/item-details/item-details-v2.component.html:
```diff
@@ -16,17 +16,22 @@ <h2 bitTypography="h6">{{ "itemDetails" | i18n }}</h2>
+          class="tw-pt-1"
+          appOrgIcon
+          [tierType]="organization.productTierType"
+          [size]="orgIconSize"
```
❓ Can we just use `size="large"` instead of creating a separate `orgIconSize` property?

shane-melton commented on libs/vault/src/components/org-icon.directive.ts:
🎉 Nice!

shane-melton commented on apps/browser/src/vault/popup/components/vault-v2/vault-list-items-container/vault-list-items-container.component.html:
```diff
@@ -27,9 +27,10 @@ <h2 bitTypography="h6">
-          [ngClass]="cipher.orgIcon"
+          appOrgIcon
+          [tierType]="cipher.organization.productTierType"
+          [size]="orgIconSize"
```
❓ Same here with just using `size="large"`.

shane-melton commented on apps/browser/src/vault/popup/views/popup-cipher.view.ts:
👍 Thanks for updating this as well!

shane-melton reviewed (commented):
Something still seems a off with the screen reader for these icons / lists. We might not be able to use both `aria-hidden` and `aria-label` at the same time. Here's an example of the VoiceOver with this PR:

https://github.com/user-attachments/assets/2de7ce64-37ff-4ca4-9ca1-33307adcf78b

It's not reading the Collection / Owner / Folder labels. (Side note: Should we make all of these items `<li>` tags so they're consistent for screen readers.)

@danielleflinn Do you have any guidance on what this should look like for screen readers?

danielleflinn commented:
@shane-melton `aria-hidden` does hide the element from the screen reader. We usually want to hide icons from SRs since most of the time they are decorative, but here they do provide meaning on the page so we need to provide the SR with an equivalent meaning. 

I originally thought we could do this by using `aria-label` (no 'aria-hidden') but after refreshing myself with [aria docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label), it seems those are only supposed to be used on interactive elements. 

What we could do instead is keep the `aria-hidden` element and then add a visually hidden element (maybe a `span`?), around the "Owner", "Folder", "Collection" text. We might also consider including a `title` attribute. 

Hope this helps!

Jingo88 commented:
@danielleflinn So I got the SR close to the example you have in your comment. For this I updated the html to use `<li>` with a sublist in Collections and added the `aria-label` to each list item. I had to make both `<i>` and `<span>` hidden so the SR wouldn't repeat. I added a recording below, let me know if this works. Also you mentioned a `title` attribute, do we still want that? if so what should they say?

https://github.com/user-attachments/assets/11fd0db1-54e8-41f4-95cc-148b4551c985

danielleflinn commented:
@Jingo88 this looks amazing!!! Thank you for your thoroughness with this 🎉 

Yes lets add the title attributes to the icons. These shouldn't impact the SR experience but they will help users if they do not understand what the icons mean. Let's add one to each of the icons with "Owner", "Collection", and "Folder" as the value.

shane-melton commented on libs/vault/src/cipher-view/item-details/item-details-v2.component.ts:
```diff
@@ -12,7 +12,7 @@ import {
   TypographyModule,
 } from "@bitwarden/components";
-
+import { OrgIconDirective } from "@bitwarden/vault";
```
⚠️ We should avoid importing `@bitwarden/vault` from within `/libs/vault`

shane-melton reviewed (changes requested):
Thanks for all the accessibility effort on this, it looks great! 🎉 

Sorry for one last import nit/issue

shane-melton reviewed (approved):
Awesome job! LGTM!
